server/auth/registrar.go

```diff
@@ -25,7 +27,320 @@ var (
 	txWaitExpiration = time.Minute
 )
 
+// bondKey creates a unique map key for a bond by its asset ID and coin ID.
+func bondKey(assetID uint32, coinID []byte) string {
+	return string(append(encode.Uint32Bytes(assetID), coinID...))
+}
+
+func (auth *AuthManager) registerBondWaiter(key string) bool {
+	auth.bondWaiterMtx.Lock()
+	defer auth.bondWaiterMtx.Unlock()
+	if _, found := auth.bondWaiterIdx[key]; found {
+		return false
+	}
+	auth.bondWaiterIdx[key] = struct{}{}
+	return true
+}
+
+func (auth *AuthManager) removeBondWaiter(key string) {
+	auth.bondWaiterMtx.Lock()
+	delete(auth.bondWaiterIdx, key)
+	auth.bondWaiterMtx.Unlock()
+}
+
+// handlePostBond handles the 'postbond' request.
+//
+// The request payload includes the user's account public key and the serialized
+// bond post transaction itself (not just the txid), the bond output's redeem
+// script, and a signature of the account ID with the key to which the bond
+// redeem script pays to demonstrate ownership of the bond by the account.
+//
+// The parseBondTx function is used to validate the transaction, and extract
+// bond details (amount, locktime, pubkey, etc.) and the account ID to which it
+// commits. This also checks that the account commitment corresponds to the
+// user's public key provided in the payload. If these requirements are
+// satisfied, the client will receive a PostBondResult in the response.
+//
+// Once the raw transaction is validated, the relevant asset node is used to
+// locate the transaction on the network.
+//
+//   - If it is not found, a PostBondResult is sent immediately with
+//     Confs=-1 set.  The bond is NOT stored in the DB. This was a courtesy
+//     pre-validation. The user should then broadcast the transaction, wait for
+//     the required number of confirmations, and send 'postbond' again.
```

Comment:
Why have a single route perform two rather different functions? Wouldn't it be more clear to have a separate route for pre-validation?

Reply:
Have given this honest consideration, and I honestly don't think so.  The machinery of the handlers would be virtually identical and whether signed or unsigned, your intend is still to *post bond*.  One less route sounds appealing to me, hence this approach.  I won't die on this hill, but I just don't feel particularly convinced or motivated to add a route.

Reply:
Still thinking this over. One thing that stands out to me is the way we're using `PostBond`

```Go
type PostBond struct {
	Signature
	AcctPubKey Bytes  `json:"acctPubKey"` // acctID = blake256(blake256(acctPubKey))
	AssetID    uint32 `json:"assetID"`
	BondVer    uint16 `json:"bondVersion"`
	BondTx     Bytes  `json:"bondTx"`
	BondScript Bytes  `json:"bondScript"`
}
```

I think that structure would make sense for pre-validation, but for actually posting the bond, it would be much more "on-model" for us to do 

```Go
type PostBond struct {
	Signature
	AssetID    uint32 `json:"assetID"`
	CoinID     Bytes  `json:"coinID"` // txid
}
```

since we can pull all of the bond details from the authoritative on-chain record. I'd almost say it's been an unwritten rule for us that we send only the minimum information required to perform the DEX functions. I've referred to this before as "minimizing the API", and I do think it's important. For one, it prevents mistakes. We can't accidentally trust what the user says. Less importantly, it reduces bandwidth.

Anyway. We can chat more here or in Element.

Reply:
I see where you're going with that.  Indented BondVer still might be important there, esp. if it indicates which contract to check (e.g. eth), but that is more on-pattern.

Reply:
Suggested changes made, routes separate.